src/tree_store/page_store/cached_file.rs

```diff
@@ -291,6 +290,38 @@ impl PagedCachedFile {
         for (offset, buffer) in write_buffer.low_pri_cache.iter() {
             self.file.write(*offset, buffer.as_ref().unwrap())?;
         }
+        for (offset, buffer) in write_buffer.cache.iter_mut() {
+            let buffer = buffer.take().unwrap();
+            let cache_size = self
+                .read_cache_bytes
+                .fetch_add(buffer.len(), Ordering::AcqRel);
+
+            if cache_size + buffer.len() <= self.max_read_cache_bytes {
+                let cache_slot: usize = (offset % Self::lock_stripes()).try_into().unwrap();
+                let mut lock = self.read_cache[cache_slot].write().unwrap();
+                lock.insert(*offset, buffer.clone(), CachePriority::High);
```

Comment:
Is it necessary to call `.clone()` here? It looks like it should be possible to just pass `buffer`

Reply:
No, that is a left-over from doing this in a single pass via `.iter`. Will fix both cases.